router: intermittent TestAddHTTPRoute tls oversized record error

I've only seen this once.

```text
----------------------------------------------------------------------
FAIL: http_test.go:112: S.TestAddHTTPRoute

http_test.go:126:
    assertGet(c, "https://"+l.TLSAddr, "example.com", "1")
http_test.go:175:
    c.Assert(err, IsNil)
... value *url.Error = &url.Error{Op:"Get", URL:"https://127.0.0.1:59890", Err:(*errors.errorString)(0xc2080f09e0)} ("Get https://127.0.0.1:59890: tls: oversized record received with length 20527")

2015/01/30 04:08:59 router: proxy error: read tcp 127.0.0.1:60732: connection reset by peer
2015/01/30 04:09:00 router: proxy error: read tcp 127.0.0.1:49394: connection reset by peer
2015/01/30 04:09:01 router: proxy error: router: no backends available
2015/01/30 04:09:01 router: proxy error: router: no backends available
2015/01/30 04:09:01 router: proxy error: router: no backends available
OOPS: 33 passed, 1 FAILED
--- FAIL: Test (6.30s)
```

https://s3.amazonaws.com/flynn-ci-logs/20150130040806-51e23402-build-b1b779ef8442583924a00b28ca806b06d4b382e8-2015-01-30-04-09-13.txt

/cc @benburkert @bgentry

--- a/router/http.go
+++ b/router/http.go
@@ -13,7 +13,6 @@ import (
 	"sync"
 	"time"
 
-	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/kavu/go_reuseport"
 	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
 	"github.com/flynn/flynn/discoverd/client"
 	"github.com/flynn/flynn/pkg/ctxhelper"
@@ -267,7 +266,7 @@ func (h *httpSyncHandler) Remove(id string) error {
 
 func (s *HTTPListener) listenAndServe() error {
 	var err error
-	s.listener, err = reuseport.NewReusablePortListener("tcp4", s.Addr)
+	s.listener, err = listenFunc("tcp4", s.Addr)
 	if err != nil {
 		return err
 	}
@@ -301,7 +300,7 @@ func (s *HTTPListener) listenAndServeTLS() error {
 		Certificates:   []tls.Certificate{s.keypair},
 	})
 
-	l, err := reuseport.NewReusablePortListener("tcp4", s.TLSAddr)
+	l, err := listenFunc("tcp4", s.TLSAddr)
 	if err != nil {
 		return err
 	}

--- a/router/server.go
+++ b/router/server.go
@@ -42,6 +42,8 @@ func (s *Router) Start() error {
 	return nil
 }
 
+var listenFunc = reuseport.NewReusablePortListener
+
 func main() {
 	defer shutdown.Exit()
 
@@ -137,7 +139,7 @@ func main() {
 		shutdown.Fatal(err)
 	}
 
-	listener, err := reuseport.NewReusablePortListener("tcp4", *apiAddr)
+	listener, err := listenFunc("tcp4", *apiAddr)
 	if err != nil {
 		shutdown.Fatal(err)
 	}

--- a/router/setup_test.go
+++ b/router/setup_test.go
@@ -3,6 +3,7 @@ package main
 import (
 	"fmt"
 	"io"
+	"net"
 	"os"
 	"testing"
 	"time"
@@ -19,6 +20,7 @@ import (
 
 func init() {
 	testMode = true
+	listenFunc = net.Listen
 }
 
 type discoverdClient interface {

--- a/router/tcp.go
+++ b/router/tcp.go
@@ -8,7 +8,6 @@ import (
 	"strconv"
 	"sync"
 
-	"github.com/flynn/flynn/Godeps/_workspace/src/github.com/kavu/go_reuseport"
 	"github.com/flynn/flynn/Godeps/_workspace/src/golang.org/x/net/context"
 	"github.com/flynn/flynn/router/proxy"
 	"github.com/flynn/flynn/router/types"
@@ -113,7 +112,7 @@ func (l *TCPListener) Start() error {
 
 	if l.startPort != 0 && l.endPort != 0 {
 		for i := l.startPort; i <= l.endPort; i++ {
-			listener, err := reuseport.NewReusablePortListener("tcp4", fmt.Sprintf("%s:%d", l.IP, i))
+			listener, err := listenFunc("tcp4", fmt.Sprintf("%s:%d", l.IP, i))
 			if err != nil {
 				l.Close()
 				return err
@@ -278,7 +277,7 @@ func (r *tcpRoute) Serve(started chan<- error) {
 	var err error
 	// TODO: close the listener while there are no backends available
 	if r.l == nil {
-		r.l, err = reuseport.NewReusablePortListener("tcp4", r.addr)
+		r.l, err = listenFunc("tcp4", r.addr)
 	}
 	started <- err
 	if err != nil {

